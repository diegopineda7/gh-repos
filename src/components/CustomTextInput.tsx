import { ErrorMessage, useField } from 'formik';

interface Props {
  label: string;
  name: string;
  type?: 'text' | 'email' | 'password';
  placeholder?: string;
}

export const CustomTextInput = (props: Props) => {
  const [field] = useField(props);
  const { name, label, type, placeholder } = props;

  return (
    <>
      <label htmlFor={name}>{label}</label>
      <input
        className="text-input"
        type={type || 'text'}
        placeholder={placeholder}
        {...field}
      />
      <ErrorMessage name={name} component="span" />
    </>
  );
};
