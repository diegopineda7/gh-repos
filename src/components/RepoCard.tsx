import { Repo } from '../interfaces';
import heartFull from '../assets/heart-full.svg';
import heartEmpty from '../assets/heart-empty.svg';
import { capitalizeFirstLetter } from '../api/utils';

interface Props {
  repo: Repo;
  isFavorite?: boolean;
  onClickFavorite: () => void;
}

export const RepoCard = ({ repo, isFavorite, onClickFavorite }: Props) => {
  return (
    <div className="repo-card">
      <div className="repo-card-header">
        <a href={repo.url} target="_blank" rel="noreferrer">
          {repo.name}
        </a>
        <img
          src={isFavorite ? heartFull : heartEmpty}
          onClick={onClickFavorite}
          alt={isFavorite ? 'Remove from favorites' : 'Mark as favorite'}
          title={isFavorite ? 'Remove from favorites' : 'Mark as favorite'}
          width={30}
        ></img>
      </div>

      <div className="repo-card-footer">
        <p>{repo.primaryLanguage.name}</p>
        <p>{capitalizeFirstLetter(repo.visibility)}</p>
      </div>
    </div>
  );
};
