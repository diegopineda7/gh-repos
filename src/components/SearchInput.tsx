interface Props {
  value: string;
  placeholder?: string;
  onChange: (name: string) => void;
  onClear: () => void;
}

export const SearchInput = ({
  value,
  placeholder = '',
  onChange,
  onClear,
}: Props) => {
  return (
    <div className="search-container">
      <input
        value={value}
        placeholder={placeholder}
        onChange={(val) => onChange(val.target.value)}
        className={'text-input search-input'}
      />
      {value && <p onClick={onClear}>X</p>}
    </div>
  );
};
