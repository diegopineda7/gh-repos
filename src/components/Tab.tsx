interface Props {
  isActive: boolean;
  label: string;
  onClick: () => void;
}

export const Tab = ({ isActive, label, onClick }: Props) => {
  return (
    <div className={`tab ${isActive ? 'tab-active' : ''}`} onClick={onClick}>
      <label>{label}</label>
    </div>
  );
};
