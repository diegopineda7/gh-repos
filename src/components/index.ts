export { CustomTextInput } from './CustomTextInput';
export { SearchInput } from './SearchInput';
export { Tab } from './Tab';
export { RepoCard } from './RepoCard';
