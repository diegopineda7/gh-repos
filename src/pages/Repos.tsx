import '../styles/styles.css';

export const ReposPage = () => {
  return (
    <div>
      <h1>My repos</h1>
    </div>
  );
};
