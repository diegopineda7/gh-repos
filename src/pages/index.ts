export { SignUpPage } from './SignUp';
export { ReposPage } from './Repos';
export { LogInPage } from './LogIn';
