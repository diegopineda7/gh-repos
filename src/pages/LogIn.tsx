import { Formik, Form } from 'formik';
import * as Yup from 'yup';

// Components
import { CustomTextInput } from '../components';

// Hooks
import useAuth from '../hooks/useAuth';
import { REQUIRED_FIELD } from '../constants';

export const LogInPage = () => {
  const { userError, validateUser } = useAuth();

  const validationSchema = Yup.object({
    username: Yup.string().required(REQUIRED_FIELD),
  });

  return (
    <div>
      <h1>Log in</h1>

      <Formik
        initialValues={{
          username: '',
        }}
        onSubmit={validateUser}
        validationSchema={validationSchema}
      >
        {({ isValid, dirty }) => (
          <Form>
            <CustomTextInput
              label="GitHub username"
              name="username"
              placeholder="myuser"
            />

            <button type="submit" disabled={!isValid || !dirty}>
              Log in
            </button>
            <h4>{userError}</h4>
          </Form>
        )}
      </Formik>
    </div>
  );
};
