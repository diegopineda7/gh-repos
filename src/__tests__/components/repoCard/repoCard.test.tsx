import { RepoCard } from '../../../components';
import renderer from 'react-test-renderer';
import { SignUpPage } from '../../../pages';
import React from 'react';

describe('Test component RepoCard', () => {
  it('render component <RepoCard />', () => {
    const tree = renderer
      .create(
        <RepoCard
          onClickFavorite={() => undefined}
          repo={{
            id: '123',
            name: 'Repo Test',
            url: 'url',
            visibility: 'PRIVATE',
            primaryLanguage: { name: 'lang' },
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
