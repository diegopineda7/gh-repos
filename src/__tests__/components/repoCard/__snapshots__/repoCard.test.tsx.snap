// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component RepoCard render component <RepoCard /> 1`] = `
<div
  className="repo-card"
>
  <div
    className="repo-card-header"
  >
    <a
      href="url"
      rel="noreferrer"
      target="_blank"
    >
      Repo Test
    </a>
    <img
      alt="Mark as favorite"
      onClick={[Function]}
      src={{}}
      title="Mark as favorite"
      width={30}
    />
  </div>
  <div
    className="repo-card-footer"
  >
    <p>
      lang
    </p>
    <p>
      PRIVATE
    </p>
  </div>
</div>
`;
